gateway/handle_http_auth.go

```diff
@@ -36,10 +36,26 @@ func (gw *Handle) writeKeyAuth(delegate http.HandlerFunc) http.HandlerFunc {
 		}
 		arctx := gw.authRequestContextForWriteKey(writeKey)
 		if arctx == nil {
+			stat := gwstats.SourceStat{
```

Comment:
can we handle this stat centrally, eg. similar to what gw#handleHttpError does?

Reply:
Addressed this... let me know if this looks good